taqueria-plugin-smartpy/common.ts

```diff
@@ -17,15 +16,46 @@ export type IntersectionOpts = CompileOpts & TestOpts;
 
 type UnionOpts = CompileOpts | TestOpts;
 
+// Should point to the latest stable version, so it needs to be updated as part of our release process.
+const SMARTPY_DEFAULT_VERSION = 'v0.16.0';
+
+const SMARTPY_VERSION_ENV_VAR = 'TAQ_SMARTPY_VERSION';
+
 const SMARTPY_ARTIFACTS_DIR = '.smartpy';
 
-// Should point to the latest version, so it needs to be updated as part of our release process.
-// Currently this points to v0.15.0
-const SMARTPY_INSTALLER =
-	'https://smartpy.io/releases/20221026-28e8c18e46035c353804eb5fd725573c5d434e8a/cli/install.sh';
+const smartpyVersionToInstallerMap: { [k: string]: string } = {
+	'v0.16.0': 'https://smartpy.io/releases/20221215-8f134ebb649f5a7b37c44fca8f336f970f523565/cli/install.sh',
+	'v0.15.0': 'https://smartpy.io/releases/20221026-28e8c18e46035c353804eb5fd725573c5d434e8a/cli/install.sh',
+	'v0.14.0': 'https://smartpy.io/releases/20220926-1c748c4572188f65a525792468e37da2182f18a2/cli/install.sh',
+};
+
+const getSmartpyVersion = (): string => {
+	const userDefinedSmartpyVersion = process.env[SMARTPY_VERSION_ENV_VAR];
+	if (userDefinedSmartpyVersion) {
+		if (smartpyVersionToInstallerMap[userDefinedSmartpyVersion]) {
+			return userDefinedSmartpyVersion;
+		} else {
+			sendWarn(
+				`Version ${userDefinedSmartpyVersion} is not supported by Taqueria yet. Will default to ${SMARTPY_DEFAULT_VERSION}`,
```

Comment:
Here it would be nice to tell the user which versions are supported, so they can make a better decision.

Reply:
Ok will do!